### PR TITLE
spirv-tools: 2020.2 -> 2021.3

### PR DESCRIPTION
**Pull request by jansol**
###### Motivation for this change

https://github.com/KhronosGroup/SPIRV-Tools/blob/master/CHANGES

Adds support for vendor-neutral ray tracing extensions and integer dot products, among other things.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Ralith — APPROVED** [review]

File: pkgs/development/libraries/spirv-headers/default.nix [hunk 1]
```diff
@@ -2,13 +2,13 @@
 
 stdenv.mkDerivation rec {
   pname = "spirv-headers";
-  version = "1.5.3";
+  version = "unstable-2021-08-11";
```
Comment (7c6f434c) [inline pkgs/development/libraries/spirv-headers/default.nix:5]: (makes me long for the time when NixOS/RFCs#107 is finalised, but it seems to be a package only used for one rev-dependency and almost never installed directly)

**Comment by jansol** [comment]
Thanks!

Can't say I'm too happy about spirv-tools' way of depending on a very specific commit (it doesn't build with the latest commit of spirv-headers anymore). But I guess the closest alternative would be just vendoring the headers in. Which I guess is what most dependents actually do. Maybe it would make sense to eventually do something similar here as well.

**Comment by 7c6f434c** [comment]
I am fine with depending on specific commit, the question is what is a way of assigning a snapshot version to it.

**Comment by Ralith** [comment]
Yeah, ultimately this is on upstream for not bothering to have/use well-defined releases.

**Pull request merged**